### DISCUSSION TITLE
ID is a field name

### DISCUSSION THREAD

**Discussion by SABKraken** [infinum/datx#1216] [Q&A]
Hi there,

one of the field in the request from the database/table is not a fixed field name like _id_ or _name_, but a **dynamic** id.
For example:
`
"12345": {
  "firstname": "Nik",
  "givenname": "Johnson"
}
`
"12345" <== the dynamic id
In the next reuqest it could be "23456" or so...

Q: How i must declare the `@Field()` for the Model in Angular for this case?

Thanks in advance  and kind regards

SAB Kraken

**Comment by DarkoKukovec** [comment]
Hi @SABKraken 
This won't work directly in datx. The only thing that comes to mind is to map this to a fixed property:
```typescript
@Field({ parse: (_, data) => data['12345'] })
public someField!: MyType;
```
